Support for new JDBC connector Snowflake (#4)
IBM/watsonx-data: connectors/presto/readme.md

systems like Snowflake and Hive, or between two different Snowflake accounts.

### Connector Configuration

To configure the Snowflake connector, create a catalog properties file in *etc/catalog* named, for example, *snowflake.properties*, to mount the Snowflake connector as the snowflake catalog. Create the file with the following contents, replacing the connection properties as appropriate for your setup:

```
connector.name=snowflake
connection-url=jdbc:snowflake://<account_identifier>.snowflakecomputing.com/?db={dbname}  
connection-user={username}
connection-password={password}
```

The `connection-url` defines the connection information and parameters to pass to the Snowflake JDBC driver.

Syntax: `jdbc:snowflake://<account_identifier>.snowflakecomputing.com/?<connection_params>`

User has to provide only the `account_identifier`, and shouldn't include the prefix `.snowflakecomputing.com`

The `connection-user` and `connection-password` are typically required and determine the user credentials for the connection.

### SSL Configuration

By default, connections to Snowflake use SSL.

### Integrating the Snowflake Plugin to Presto   

> 1. Untar *watsonx-data/connectors/presto/snowflake_tarball/presto-snowflake-0.288/snowflake.tar* and place the plugin in the production’s plugins directory(presto-server-<version>/plugin).

Reviewer: Should we instead include instructions to build this project, and retrieve it from the project's target directory?  This tar will become out of date immediately after the next PR is merged.

### Connector Configuration

To configure the Snowflake connector, create a catalog properties file in *etc/catalog* named, for example, *snowflake.properties*, to mount the Snowflake connector as the snowflake catalog. Create the file with the following contents, replacing the connection properties as appropriate for your setup:

```
connector.name=snowflake
connection-url=jdbc:snowflake://<account_identifier>.snowflakecomputing.com/?db={dbname}  
connection-user={username}
connection-password={password}
```

The `connection-url` defines the connection information and parameters to pass to the Snowflake JDBC driver.

Syntax: `jdbc:snowflake://<account_identifier>.snowflakecomputing.com/?<connection_params>`

User has to provide only the `account_identifier`, and shouldn't include the prefix `.snowflakecomputing.com`

The `connection-user` and `connection-password` are typically required and determine the user credentials for the connection.

### SSL Configuration

By default, connections to Snowflake use SSL.

### Integrating the Snowflake Plugin to Presto (Local Setup)

1. Clone the repository [watsonx-data](https://github.com/IBM/watsonx-data).

> 2. Copy the directory *watsonx-data/connectors/presto/presto-snowflake* to the *presto[presto-root]* directory.

Author: @tdcmeehan - I have included the necessary instructions for utilizing the Snowflake jar and integrating the plugin with Presto here: https://github.com/IBM/watsonx-data/pull/4/files#diff-f41e6bedcbfb5378f732958787ac233f6b8352eb1cca31a7897ee9bdf145779eR81. 

Could you please verify if this is sufficient?

> 2. Copy the directory *watsonx-data/connectors/presto/presto-snowflake* to the *presto[presto-root]* directory.

Reviewer: It's still not sufficient.  These instructions presume the user will have their own build of Presto, it needs to be rewritten to avoid that entirely.  It should just be as simple as building the JAR for this project and dropping it into the Presto deployment's plugins directory (and adding the configuration).

    ```
    <fileSet>
        <directory>${project.build.directory}/dependency/presto-snowflake-${project.version}</directory>
        <outputDirectory>plugin/snowflake</outputDirectory>
    </fileSet>
    ```

6.  Add dependency on the module in *presto-server/pom.xml*:  

    ```
    <dependency>
            <groupId>com.facebook.presto</groupId>
            <artifactId>presto-snowflake</artifactId>
            <version>${project.version}</version>
            <type>zip</type>
            <scope>provided</scope>
    </dependency>
    ```

7. Run the following command from the project root directory:

    `./mvnw clean install -DskipTests`

8. Create the file `snowflake.properties` at the following path: `presto-main/etc/catalog/snowflake.properties` replacing the properties as appropriate.

9. Append the property `allow-drop-table` to `true` in `presto-main/etc/catalog/snowflake.properties` to allow dropping Snowflake tables from Presto via `DROP TABLE` (defaults to `false`).

10. Run the Presto server:

    `./bin/launcher start`

### Integrating the Snowflake Plugin to Presto (Production Setup)    

> 1. Untar *watsonx-data/connectors/presto/snowflake_tarball/presto-snowflake-0.288/snowflake.tar* and place the plugin in the production’s plugins directory.

Reviewer: Can you specify the directory?

> 2. Create the file `snowflake.properties` at the following path: `presto-main/etc/catalog/snowflake.properties` replacing the properties as appropriate.   

Reviewer: ```suggestion
2. Create the file `snowflake.properties` at the following path: `tc/catalog/snowflake.properties` replacing the properties as appropriate.   
```

> 3. Append the property `allow-drop-table` to `true` in `presto-main/etc/catalog/snowflake.properties` to allow dropping Snowflake tables from Presto via `DROP TABLE` (defaults to `false`).

Reviewer: ```suggestion
3. Append the property `allow-drop-table` to `true` in `etc/catalog/snowflake.properties` to allow dropping Snowflake tables from Presto via `DROP TABLE` (defaults to `false`).
```

4. Run the Presto server:

    `./bin/launcher start`

### Multiple Snowflake Databases or Accounts

The Snowflake connector can only access a single database within
a Snowflake account. Thus, if you have multiple Snowflake databases,
or want to connect to multiple Snowflake accounts, you must configure

> multiple instances of the Snowflake connector.

Reviewer: ```suggestion
multiple catalog entries of the Snowflake connector.
```
